src/firefly/js/drawingLayers/WebGrid.js

```diff
@@ -0,0 +1,882 @@
+
+}
+
+ /**
+  * Implement this method defined in DrawLayerFactory.js
+  * @param dataType
+  * @param plotId
+  * @param drawLayer
+  * @param action
+  * @param lastDataRet
+  * @returns {*}
+  */
+
+function getDrawData(dataType, plotId, drawLayer, action, lastDataRet){
+
+     var plot= primePlot(visRoot(),plotId);
+     if (!plot)return null;
+
+     var cc= CsysConverter.make(plot);
+     if (!cc) return null;
+
+     var drawDataArray= getData(plot, cc);
+     return drawDataArray|| lastDataRet;
+ }
+
+ /**
+  * Return a Coordinate object for a given coordinate name
+  * @param csysName
+  * @returns {*}
+  */
+
+ function getCoordinateSystem(csysName) {
+
+     for (let i = 0; i < coordinateArray.length; i += 1) {
+         if (get(coordinateArray[i], 'coordName') === csysName) {
+             return coordinateArray[i].csys;
+
+         }
+     }
+
+ }
+
+ /**
+  * This method does the calculation for drawing data array
+  * @param plot
+  * @param cc
+  */
+ function getData (plot,  cc){
+
+     const {width, height, screenWidth, csys,labelFormat} = getDrawLayerParameters(plot);
+     if (width > 0 && height >0) {
+         var bounds = new Rectangle(0, 0, width, height);
+         var factor =  plot.zoomFactor;
+         if (factor < 1.0) factor = 1.0;
+         var range = getRange(csys, width, height, cc);
+         var levels = getLevels(range, factor);
+         var labels = getLabels(levels, csys, labelFormat);
+         var {xLines, yLines} = computeLines(cc, csys, range, levels, screenWidth);
+         var wpt = cc.getWorldCoords(makeImageWorkSpacePt(1, 1), csys);
+         var aitoff = (!wpt);
+         var drawDataArray = drawLines(csys, bounds, labels, xLines, yLines, levels[0].length, levels[1].length, aitoff, screenWidth);
+         return drawDataArray;
+     }
+ }
+ /**
+  * This method prepare the input parameters needed for calculating the drawing data array
+  * @param plot
+  * @returns {{width: (dataWidth|*), height: (*|dataHeight), screenWidth: *, csys: *, labelFormat: string}}
+  */
+ function getDrawLayerParameters(plot){
+    var width = plot.dataWidth;
+    var height = plot.dataHeight;
+    var screenWidth = plot.screenSize.width;
+
+    var csysName = AppDataCntlr.getPreference(COORDIANTE_PREFERENCE);
+    if (!csysName) {
+        csysName='eq2000hms';//set default
+    }
+    var csys=getCoordinateSystem(csysName);
+
+
+    var labelFormat=csysName.endsWith('hms')? 'hms':'dcm';
+
+    return {width, height, screenWidth, csys,labelFormat};
+}
+ /**
+  * Implement this one defined in DrawLayerFactory.js to handle the layer changes when the coodinate property is changed.
+  * @param drawLayer
+  * @param action
+  * @returns {null}
+  */
+ function getLayerChanges(drawLayer, action) {
+
+
+    var {plotIdAry}= action.payload;
+    var plot= primePlot(visRoot(),plotIdAry[0]);
+
+
+    if (!plot)return null;
+    var cc= CsysConverter.make(plot);
+    if (!cc) return null;
+
+
+    return getData(plot, cc);
```

Comment:
Doing this is very bad for performances.  You want to use only getDrawData for returning the grid data. Get layer changes should only be reserved for simple layers or changing other attributes of the drawing layers. In general, if this function returns anything that is not null, the drawing layer will update.